--- origsrc/libkate-0.1.5/Makefile	2008-07-07 14:36:35.000000000 -0500
+++ src/libkate-0.1.5/Makefile	2008-07-17 21:53:25.609375000 -0500
@@ -60,16 +60,20 @@
 
 OBJS_STATIC=$(foreach module, $(MODULES),$(OBJDIR)/static/$(module).o)
 OBJS_SHARED=$(foreach module, $(MODULES),$(OBJDIR)/shared/$(module).o)
+OBJS_LIBTOOL=$(foreach module, $(MODULES),$(OBJDIR)/$(module).lo)
 
 LIBOGGKATE_OBJS_STATIC=$(OBJDIR)/static/kate_ogg.o
 LIBOGGKATE_OBJS_SHARED=$(OBJDIR)/shared/kate_ogg.o
+LIBOGGKATE_OBJS_LIBTOOL=$(OBJDIR)/kate_ogg.lo
 
 STATICLIBS=$(LIBDIR)/liboggkate.a $(LIBDIR)/libkate.a
 SHAREDLIBS=$(LIBDIR)/liboggkate.$(VERSION).so $(LIBDIR)/libkate.$(VERSION).so
+LIBTOOLLIBS=$(LIBDIR)/liboggkate.la $(LIBDIR)/libkate.la
 
 all: staticlib sharedlib tools #doc
 staticlib: $(STATICLIBS)
 sharedlib: $(SHAREDLIBS)
+libtoollib: $(LIBTOOLLIBS)
 
 LEX=$(shell which flex 2> /dev/null)
 YACC=$(shell which bison 2> /dev/null)
@@ -107,6 +111,11 @@
 	@mkdir -p $(dir $@)
 	@$(CC) $(CFLAGS_SHARED) $(CWARNFLAGS_FULL) -o $@ -c $<
 
+$(OBJDIR)/%.lo: src/%.c
+	@echo " CC $@"
+	@mkdir -p $(dir $@)
+	@libtool --silent --mode=compile --tag=CC $(CC) $(CFLAGS) -o $@ -c $<
+
 $(OBJDIR)/%.o: tools/%.c
 	@echo " CC $@"
 	@mkdir -p $(dir $@)
@@ -162,6 +171,16 @@
 	@/sbin/ldconfig -n $(LIBDIR)
 	@ln -fs liboggkate.so.$(SONAME_MAJOR) lib/liboggkate.so
 
+$(LIBDIR)/libkate.la: $(OBJS_LIBTOOL)
+	@echo " LD $@"
+	@mkdir -p $(dir $@)
+	@libtool --silent --mode=link --tag=CC $(CC) -rpath $(PREFIX)/lib -version-info $(SONAME_MAJOR) -no-undefined -o $@ $^
+
+$(LIBDIR)/liboggkate.la: $(LIBOGGKATE_OBJS_LIBTOOL) $(LIBDIR)/libkate.la
+	@echo " LD $@"
+	@mkdir -p $(dir $@)
+	@libtool --silent --mode=link --tag=CC $(CC) -rpath $(PREFIX)/lib -version-info $(SONAME_MAJOR) -no-undefined -o $@ $^
+
 .PHONY: clean
 clean:
 	rm -f $(OBJS_STATIC) $(OBJS_STATIC:.o=.d)
@@ -196,6 +215,21 @@
            awk -v px="$(PREFIX)" '/^prefix=/ {print "prefix="px; next} {print}' \
            > $(PREFIX)/lib/pkgconfig/oggkate.pc
 
+.PHONY: install-libtool
+install-libtool: libtoollib
+	mkdir -p $(DESTDIR)$(PREFIX)/include/kate
+	install -m644 include/kate/kate.h include/kate/oggkate.h $(DESTDIR)$(PREFIX)/include/kate/
+	mkdir -p $(DESTDIR)$(PREFIX)/lib
+	libtool --silent --mode=install install $(LIBDIR)/libkate.la $(DESTDIR)$(PREFIX)/lib/
+	libtool --silent --mode=install install $(LIBDIR)/liboggkate.la $(DESTDIR)$(PREFIX)/lib/
+	mkdir -p $(DESTDIR)$(PREFIX)/lib/pkgconfig
+	cat misc/pkgconfig/kate.pc\
+           | awk -v px="$(PREFIX)" '/^prefix=/ {print "prefix="px; next} {print}' \
+           > $(DESTDIR)$(PREFIX)/lib/pkgconfig/kate.pc
+	cat misc/pkgconfig/oggkate.pc | \
+           awk -v px="$(PREFIX)" '/^prefix=/ {print "prefix="px; next} {print}' \
+           > $(DESTDIR)$(PREFIX)/lib/pkgconfig/oggkate.pc
+
 .PHONY: uninstall
 uninstall:
 	-rm -f $(PREFIX)/lib/libkate.a $(PREFIX)/lib/liboggkate.a
@@ -300,7 +334,6 @@
 	  cmp $(tmp_kate1) $(tmp_kate2) && \
 	) rm -f $(tmp_ogg1) $(tmp_ogg2) $(tmp_kate1) $(tmp_kate2)
 
-tools/encoder tools/decoder: $(LIBDIR)/liboggkate.a $(LIBDIR)/libkate.a
 
 ifneq '$(findstring clean,$(MAKECMDGOALS))' 'clean'
 -include obj/*.d obj/*/*.d
